### DISCUSSION TITLE
RectangularSection with DateTimeAxis

### DISCUSSION THREAD

**Discussion by beardboy** [Live-Charts/LiveCharts2#1526] [Q&A]
I am trying to create a RectangularSection while using a DateTimeAxis. I assumed I would want to get the DateTime.Ticks from the DateTimePoint to find my Xi and Xj points. The red dash is showing up on the Y axis as specified, however nothing is showing up with the X rectangle so I'm not sure if my assumptions were correct. Any help would be appreciated.

`

        var xi = points.FirstOrDefault(x => x.DateTime.Hour == 15 && x.DateTime.Minute == 00).DateTime.Ticks;
        var xj = points.FirstOrDefault(x => x.DateTime.Hour == 20 && x.DateTime.Minute == 00).DateTime.Ticks;

        var sections = new RectangularSection[]
        {
            new()
            {
                Yi = 2,
                Yj = 2,
                Stroke = new SolidColorPaint
                {
                    Color = SKColors.Red,
                    StrokeThickness = 3,
                    PathEffect = new DashEffect(new float[] { 6, 6 })
                }
            },
            new()
            {
                Xi = xi,
                Xj = xj,
                Fill = new SolidColorPaint { Color = SKColors.Blue.WithAlpha(20) }
            },
        };

`

**Comment by begleysm** [comment]
I was working on someone else's code (they setup the axes) but was able to make something work.

The X-Axis is setup like so
```
_customAxis = new DateTimeAxis(TimeSpan.FromSeconds(1), Formatter)
    {
        CustomSeparators = GetSeparators1(),
        AnimationsSpeed = TimeSpan.FromMilliseconds(0),
        SeparatorsPaint = new SolidColorPaint(SKColors.Black.WithAlpha(100)),
        LabelsPaint = new SolidColorPaint(SKColors.Yellow)
    };
```

Standard points are placed on the graph like this
```
var point = new ErrorDateTimePoint(timeNow, currentValue, -1000, 1000);
    if (!last_item_status)
    {
        pointsCollection.Add(point);
    }
```
                                        
And I was able to place a RectangularSection on a relevant timeline like this
```
rectangularSectionCollection.Add(
    new RectangularSection
        {
            Xi = previousTimeNow.Ticks,
            Xj = timeNow.Ticks,
            Fill = new SolidColorPaint { Color = SKColors.White },
       }
    );
```

**Comment by beardboy** [comment]
Sorry for the late response.

I was able to search for the date from my points, and then add the sections for multiple days.

```
var firstDay = points.FirstOrDefault().LocalDateTime.Date;
var lastDay = points.LastOrDefault().LocalDateTime.Date;

var sections = new List<RectangularSection>();
while (firstDay <= lastDay)
{
    sections.Add(new()
    {
        Xi = firstDay.AddHours(15).Ticks,
        Xj = firstDay.AddHours(20).Ticks,
        Yi = min,
        Yj = max,
        Fill = new SolidColorPaint { Color = SKColors.Red.WithAlpha(40) }
    });

    firstDay = firstDay.Date.AddDays(1);
}
```

<img width="1920" height="1020" alt="Screenshot 2025-07-23 1 09 15 AM" src="https://github.com/user-attachments/assets/a23fae73-6046-4b2a-a211-94f7327ce04a" />

<img width="1920" height="1020" alt="Screenshot 2025-07-23 1 08 26 AM" src="https://github.com/user-attachments/assets/c325524e-29f8-4338-8a29-8778ed91d0ab" />
